tests/net/hostname/src/main.c

```diff
@@ -310,6 +339,19 @@ static int bytes_from_hostname_unique(uint8_t *buf, int buf_len, const char *src
 	return 0;
 }
 
+static int init_event_handler(void)
+{
+	if (IS_ENABLED(CONFIG_NET_MGMT_EVENT)) {
```

Comment:
I don't think C check is ok here, given that `hostname_cb` is hidden behind the preprocessor check. Maybe the entire SYS_INIT with it's handler should be compiled out at the preprocessor level?

Reply:
Done